data/transactions/transaction.go

```diff
@@ -359,6 +359,10 @@ func (tx Transaction) WellFormed(spec SpecialAddresses, proto config.ConsensusPa
 			return fmt.Errorf("tx.GlobalStateSchema too large, max number of keys is %d", proto.MaxGlobalSchemaEntries)
 		}
 
+		if tx.ExtraProgramPages > proto.MaxExtraAppProgramPages {
```

Comment:
Very small complaint - seem like we ought to move this check up, so that tx.ExtraProgramPages is checked before using it on lines 346 and 350.  No difference in result, but it seem right.